centreon/src/Core/HostCategory/Domain/Model/HostCategory.php

```diff
@@ -45,4 +47,30 @@ public function getId(): int
     {
         return $this->id;
     }
+
+    /**
+     * @param string $name
+     */
+    public function setName(string $name): void
+    {
+        $name = trim($name);
+        $shortName = (new \ReflectionClass($this))->getShortName();
+        Assertion::maxLength($name, self::MAX_NAME_LENGTH, $shortName . '::name');
+        Assertion::notEmptyString($name, $shortName . '::name');
+
+        $this->name = $name;
+    }
+
+    /**
+     * @param string $alias
+     */
+    public function setAlias(string $alias): void
+    {
+        $alias = trim($alias);
```

Comment:
maybe use TrimmedString ?

Reply:
not in models

Reply:
The typehint `TrimmedString` is mainly used when we need to pass a string trimmed to another object because it makes sense in the functional part and we don't want to rely on the developer to think about explicitely trim the string.

In a model, this make almost no sense because the model itself knows how to store its strings via setters.